plugin/evm/limitorders/build_block_pipeline.go

```diff
@@ -63,13 +65,34 @@ func (pipeline *BuildBlockPipeline) runLiquidationsAndMatchingForMarket(market M
 	}
 	shortOrders := pipeline.db.GetShortOrders(market, shortCutOffPrice)
 
-	// 3. Run liquidations
+	// 3. Remove orders that were just cancelled
+	longOrders = removeOrdersWithIds(longOrders, cancellableOrderIds)
+	shortOrders = removeOrdersWithIds(shortOrders, cancellableOrderIds)
+
+	// 4. Run liquidations
 	modifiedLongOrders, modifiedShortOrders := pipeline.runLiquidations(market, longOrders, shortOrders)
 
-	// 4. Run matching engine
+	// 5. Run matching engine
 	runMatchingEngine(pipeline.lotp, modifiedLongOrders, modifiedShortOrders)
 }
 
+func (pipeline *BuildBlockPipeline) cancelOrders(oraclePrices map[Market]*big.Int) map[common.Hash]struct{} {
+	cancellableOrders := pipeline.db.GetOrdersToCancel(oraclePrices)
+	cancellableOrderIds := map[common.Hash]struct{}{}
```

Comment:
It's the general convention to use struct{} instead of bool. I also like it that there's no confusion with struct{} unlike with bool(when bool = false)

https://stackoverflow.com/a/37320521/2485594